app/models/user.rb

```diff
@@ -1,34 +1,63 @@
 # frozen_string_literal: true
 
 class User < ApplicationRecord
+  has_secure_password
+
+  before_create :confirmation_token
+
   require 'uri'
-  # has_secure_password
+  validates_uniqueness_of :email
+
+  before_save do
+    self.email = email.downcase
+  end
 
   validates :username,
             uniqueness: true,
             presence: true,
-            length: { minimum: 2, maximum: 20 },
-            allow_nil: false
+            length: { minimum: 2, maximum: 20 }
 
   validates :email,
-            uniqueness: true,
+            uniqueness: { case_sensitive: false },
```

Comment:
Nanu, willst du wirklich Mails nicht `uniqueness: { case_sensitive: true }` lauten?

Reply:
Ja stimmt E-Mails sind ja  case_sensitive

Reply:
mail-adressen sind im allgemeinen, afaik, nicht case_sensitive
https://stackoverflow.com/questions/9807909/are-email-addresses-case-sensitive
"[...], no widely used mail systems distinguish different addresses based on case. [...]"
jedoch:
"[...] In short, you are safe to treat email addresses as case-insensitive."
(also, man sollte es machen, man MUSS es nur nicht umbedingt)